src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketRepo.java

```diff
@@ -0,0 +1,57 @@
+package com.atlassian.bitbucket.jenkins.internal.scm;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+
+public class BitbucketRepo {
```

Comment:
Personal preference, but I think this class and `BitbucketRepoFetcher` aren't really needed. I think it could be nicely done all in one class (not necessarily BitbucketSCM, maybe a different class entirely) rather than having this object that just holds a pair of values and an interface that is essentially a function

Reply:
Let me share how many places we can use this string project and repository together, 

1. BitbucketWebhookClientImpl constructor
2. BitbucketSCMRepository constructor and instance variable
3. doCheckRepositoryName in BitbucketSCM which hopefully can be used as QueryParameter
4. WebhookRegisterRequest 
5. BitbucketRepositoryClientImpl constructor. 
6. In test code. 

I truly feel that we need this whenever we need to pass in repo slug. Using them separately, 

1. Increases the need to have builder pattern since it increases the number of parameters
2. A successful invariant of a repo slug is non-empty project key. You can only enforce this if you have them together. 

Let me remove BitbucketRepo for now. We can come back to this again.